design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+* **Stacks** will be given permissions only within a **single namespace**.
+* All stacks will be granted permissions to a small subset of core Kubernetes resources as well as access to any CRDs they own and depend on.
+
+This means that stacks should not be able to directly affect any resources outside of the specific namespace they are installed into, while cluster stacks operate at a higher level of privilege and can affect the entire cluster.
+This is a familiar concept to the Kubernetes permission scoping model of `ClusterRole` and `Role`.
+
+### CRD Scoping
+
+To avoid a common point of confusion, let's review the available scoping of CRDs.
+Custom resource **instances** can be scoped to either a single namespace or to the entire cluster.
+It's important to note that this refers to the instances, not the CRD type definitions.
+From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition):
+
+> CustomResourceDefinitions themselves are non-namespaced and are available to all namespaces
+
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
+This has caused confusion in the past, so it's important to clear that up here.
+
+## Allowed Resource Access
+
+As previously stated, Crossplane is not intended to be a typical/full Kubernetes cluster to run arbitrary workloads.
+Once installed into the control plane, stacks will mostly target external resources and a known set of CRDs that they own and depend on.
+Therefore, we can lock down the allowed permissions for stacks to a subset of all available permissions in the Kubernetes API.
+
+The `rbac.yaml` file that allows a stack to specify the permissions it requires will be removed and no longer supported.
+Instead, the permissions for a stack can be considered **implicit**.
+The following permissions will be granted to a stack when it is installed:
+
+* **Core Kubernetes API kinds:** Simple configuration and monitoring related types such as `ConfigMap`, `Secrets` and `Events`.
+There should be no need for a stack to create a Deployment, Pod, Volume, or other types like that.
+* **Owned CRDs:** All CRDs that the stack owns and are being installed by the stack.
+* **Dependent CRDs:** All CRDs that the stack directly depends on and will consume to perform its primary operations (e.g. `MySQLInstance` claim).
+
+The Stack Manager will create the necessary service account, role, and binding that includes these permissions and serves as the runtime security context that the Stack will operate in.
+This is described in greater details in the sections below.
+
+In the future, other security modes could be enabled such as a more explicit model, and the allowed set of permissions could be expanded, but we are intentionally starting now with a small scope.
```

Comment:
This seems to answer my earlier question about whether explicit permissions would be supported in the future. Would it make sense to move it closer to where the reader may have the question? Or maybe to make it more obvious by having its own section header, so that the reader knows where the question will be addressed?

Reply:
I've updated the wording here a bit and this explanation is within the same section as your previous question.  I think that will be sufficient.